File: azure-typescript/3_component-resources/storage-infra.ts
import * as storage from "@pulumi/azure-native/storage";
import { ComponentResource, ComponentResourceOptions, Input, Output } from "@pulumi/pulumi";
import * as pulumi from "@pulumi/pulumi";

export interface StorageInfraArgs {
    parameter1: Input<string>;
};

export class StorageInfra extends ComponentResource {
    public readonly output1: Output<string>;
    public readonly output2: Output<string>;
    public readonly output3: Output<string>;

    // Must have a constructor that defines the parameters and namespace - "custom:x:Storage" in this case.
    constructor(name: string, args: StorageInfraArgs, opts?: ComponentResourceOptions) {
        super("custom:x:StorageInfra", name, args, opts);

        /////
        // Declare the children resources (i.e. storageAccount) be sure to include a resource option of "parent:this".
        /////

        // End with this. It is used for display purposes.
        this.registerOutputs();
    };
};

File: azure-csharp/3_component-resources/StorageInfra.cs
////
// Component Resource Shell
// (commented out to avoid compilation errors until ready to use)
////

// using System;
// using System.Collections.Generic;
// using System.Linq;
// using System.Threading.Tasks;
// using Pulumi;
// using Pulumi.AzureNative.Storage;
// using Pulumi.AzureNative.Storage.Inputs;

// class MyStorageInfraArgs
// {
//     public Input<string> Parameter1 { get; set; } = null!;
// }

// class MyStorageInfra: Pulumi.ComponentResource
// {
//     public Output<string> Output1;
//     public Output<string> Output2;
//     public Output<string> Output3;
//     public Output<string> Output4;

//     public MyStorageInfra(string name, MyStorageInfraArgs args, ComponentResourceOptions? opts = null)
//         : base("custom:x:StorageInfra", name, opts)
//     {
//         /////
//         // Declare the children resources (i.e. storageAccount) be sure to include a resource option of "Parent = this".
//         /////

//         this.Output = OutputValue

//     }
// }